### PR TITLE
feat(va-checkbox): array access

### LINKED ISSUE #564: Checkbox should have array access
* **I'm submitting a ...**
  - [x] feature request

* **What is the motivation / use case for changing the behavior?**
It's widely useful. Especially for smaller apps.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

Interface might look like this
```html
<va-checkbox v-model="array" arrayValue="one"/>
<va-checkbox v-model="array" arrayValue="two"/>
```

- [ ] If `arrayValue` is included in array - checkbox should be selected.
- [ ] On change checkbox should emit new array with `arrayValue` added/removed.

### PR DESCRIPTION
**Pull request by serjilyashenko**
feat(va-checkbox): add array access to va-checkbox component

close #564 

## Description

- Changed va-checkbox wiki in accordance with the new functionality.
- Array access functionality implemented identical to va-toggle component.
- Added a demo of new use.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md [hunk 1]
```diff
@@ -13,13 +13,19 @@
...
+  array-value="one"
+/>
+```
 
 **Props**
 * `id` - String
 * `name` - String
 * `label` - String - label to the right of checkbox
 * `value` - Boolean - main value
+* `array` - Array<String>
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md]: No such a prop.

File: src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md [hunk 2]
```diff
@@ -13,13 +13,19 @@
...
+* `arrayValue` - String - if arrayValue is included in value array - checkbox should be selected
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md]: Probably no type validation here would make more sense.

Definitely could be string, object, symbol, number etc.

File: src/vuestic-theme/vuestic-components/va-checkbox/VaCheckbox.demo.vue [hunk 3]
```diff
@@ -85,6 +92,7 @@ export default {
   data () {
     return {
       value: true,
+      selection: [],
```
Comment (asvae) [inline src/vuestic-theme/vuestic-components/va-checkbox/VaCheckbox.demo.vue]: Does nothing.

Comment (serjilyashenko) [reply src/vuestic-theme/vuestic-components/va-checkbox/VaCheckbox.demo.vue]: This is target array for demo purposes.
It passes to va-checkbox components set.
```
    <VbCard title="Array as model">
      {{selection}}
      <va-checkbox v-model="selection" array-value='one' label="one"/>
      <va-checkbox v-model="selection" array-value='two' label="two"/>
      <va-checkbox v-model="selection" array-value='three' label="three"/>
      <va-checkbox v-model="selection" array-value='four' label="four"/>
    </VbCard>
```

File: src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md (hunk 2 above)
Comment (serjilyashenko) [reply src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md]: "Any" type can cover them all.
Is it ok?

File: src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md (hunk 1 above)
Comment (serjilyashenko) [reply src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md]: Sorry. I didn't understand correctly.
I'll fix this.

File: src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md (hunk 2 above)
Comment (asvae) [reply src/vuestic-theme/vuestic-components/va-checkbox/va-checkbox-docs.md]: I think you can just skip type.
There is no `any` in JS, and we're not going TS yet. :)

**Comment by asvae** [comment]
Thanks and congrats on first PR. :)

**Pull request merged**